### PR TITLE
Update to Pixi 6

### LINKED ISSUE #52: Add support for PIXI.js v6
Version 6 is out but it's not an automatic upgrade for this library. v6 is written in TypeScript and has much better support for types and drops some legacy support.

Also raises the question of how to version this library to work with each if they're not compatible. 🤔

### PR DESCRIPTION
**Pull request by luxaritas**
Resolves #52 

A couple notes:
* The tests are currently failing, but they were already failing. I don't believe I've caused them to fail any worse (or differently)
* The tests now have a message from PIXI complaining about WebGL being unavailable for compressed textures. This shouldn't have any actual detrimental effect, but it does mean that the logs are a bit messy. This may be resolvable by importing individual members instead of using `import *`, but I'm not sure. There is also a webgl mock for jest that I see on NPM, but it is not well maintained.
* Since IFontMetrics is no longer exported, I redefined it in types.ts and then imported it elsewhere in the code in its place. Let me know if you have a different preference.
* The `as const`s are to narrow the type of font styles, weights, etc. from `string` to the specific string literals, which exist in the type union that these properties now have in their type definition.

Let me know if you have any questions or want additional changes.

### CODE REVIEW

File: test/TaggedText.test.ts [hunk 1]
```diff
@@ -23,8 +23,8 @@ describe("TaggedText", () => {
       fontSize: 10,
       fontFamily: "arial",
     },
-    b: { fontWeight: "bold" },
-    i: { fontStyle: "italic" },
+    b: { fontWeight: "bold" as const },
```
Comment (mimshwright) [inline test/TaggedText.test.ts:26]: This is a neat trick I wasn't aware of!

**Review by mimshwright — APPROVED** [review]
LGTM

**Comment by mimshwright** [comment]
> * The tests are currently failing, but they were already failing. I don't believe I've caused them to fail any worse (or differently)
These appear to be passing for me. The only thing I'm seeing is the webgl issues. 

> * The tests now have a message from PIXI complaining about WebGL being unavailable for compressed textures. This shouldn't have any actual detrimental effect, but it does mean that the logs are a bit messy. This may be resolvable by importing individual members instead of using `import *`, but I'm not sure. There is also a webgl mock for jest that I see on NPM, but it is not well maintained.
Any ideas on whether it's possible to fix this? It kind of makes the tests unreadable. I assume this is maybe an issue with trying to render PIXI in the headless environment of the test runner? I tried googling about this some but haven't found anything so far. I'm going to add an issue for pixi just in case anyone wants to help out. 

> * Since IFontMetrics is no longer exported, I redefined it in types.ts and then imported it elsewhere in the code in its place. Let me know if you have a different preference.
Sounds good. 

> * The `as const`s are to narrow the type of font styles, weights, etc. from `string` to the specific string literals, which exist in the type union that these properties now have in their type definition.
This is fine. I think I'd prefer to type the entire style as a `TextStyle` or whatever, but this is fine. 

Will keep open for now pending a solution for the WebGL issue.

**Comment by luxaritas** [comment]
> This is fine. I think I'd prefer to type the entire style as a `TextStyle` or whatever, but this is fine.

Doh! That would be the right way to do it (or rather, `Partial<TextStyle>` or whatever is used elsewhere). For some reason that didn't cross my mind.

**Comment by mimshwright** [comment]
Looks like the solution to the testing issue might be to use https://github.com/hustcc/jest-electron or https://github.com/facebook-atom/jest-electron-runner instead of jsdom / canvas for the rendering. 

I'm trying out the first one and so far it's working ok. It kinda breaks my VSCode tests and i'm seeing pixel values that are 1px off from what they were before the change so it's still not perfect. but maybe the best possibility.

**Comment by luxaritas** [comment]
FWIW, I was getting values 1px off when I was running the tests before, so it may be related... somehow.

**Pull request merged**